io.github.syntalos.yaml

```diff
@@ -0,0 +1,244 @@
+id: io.github.syntalos
```

Comment:
Code hosting IDs needs 4 components at least https://docs.flathub.org/docs/for-app-authors/requirements#control-over-domain-or-repository

`io.github.syntalos.syntalos` would be fine to match the repo name, we automaitcally check through the linter if the repo exists by reversing the rDNS

Reply:
This rule feels a bit arbitrary - we would really like to keep the much shorter name and not have to rename the app again, as due to how Flatpak works the ID shows up in many file paths that users will have to access.

There are also many examples on Flathub that do not follow this rule, e.g. `io.github.Hexchat`, `io.github.ImEditor`, `io.github.OpenToonz` and many more...

Reply:
Those were submitted before the rule came, the issues with 3 components is explained in https://docs.flatpak.org/en/latest/conventions.html#application-ids

Reply:
This convention existed for a long time, we started enforcing it since 2022

Reply:
Okay, this is a problem actually (as silly as that sounds...) - I will have to have some conversations here to see what we can do about it.

Reply:
(With "here" I mean at my institution, not "in this issue" ^^ - either we move to a new domain or have to change the ID again, I guess)

Reply:
Preferably `io.github.syntalos.syntalos`, then linter can calculate the repo to be github.com/syntalos/syntalos and check it exists.

If you differ on the last component, then it'd need an exception.

Reply:
Curious question: What if a `vcs-browser` URL is also present in the metadata? https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-url

Will you ignore the ID then, instead of distilling meaning other than "project owns web location" from it? What happens if those two values differ and point to different VCS browser locations?

Reply:
Appstream tags aren't read for this. We verify the ID and domain/webpage manually and through linter.

Appstream tags can be changed without going through us, the ID cannot.